blockchains/eth/eth_worker.js

```diff
@@ -145,11 +146,10 @@ class ETHWorker extends BaseWorker {
     return result;
   }
 
-  async work() {
+  async work(key) {
```

Comment:
You give this 'key' argument to the `work()` method just to have it back. It does not need to know about this key, it does nothing with it. Wrap the call in a closure if you need to preserve it.

Reply:
It doesn't look really elegant atm, but I didn't think of another way to somehow set an order of the tasks beforehand
---
but I think I've thought of a way with queue.on('add'), which would be elegant

Reply:
> Wrap the call in a closure if you need to preserve it.

Wouldn't what I suggested work? I am thinking this:

```js
const key = "some key"
this.taskManager.pushToQueue(() => {
  const result = this.worker.work()
  return [key, result]
  }
);
  ```

Reply:
Hm, this is simple, yes. Thanks